src/main.rs

```diff
@@ -1,17 +1,29 @@
 mod lib;
 
 use lib::Parser;
+use lib::languages::generate;
 use std::fs::File;
 use std::io::prelude::*;
 
 fn main() {
-    // the .tp is a temporary file extension (Talpa Language)
-    // the example file should be updated with all working components
     let mut file = File::open("./src/example.tp").unwrap();
     let mut contents = vec![];
     file.read_to_end(&mut contents).unwrap();
-    match Parser::parse(contents) {
+    let parser = Parser::parse(contents);
+    match parser {
```

Comment:
Maybe this is a better way of checking for the ok value then we don't have this nesting going on here
```rust
let res = match parser {
    Err(err) => {
        println!("{}", err);
        return;
    }
    Ok(res) => res,
}
```